### PR TITLE
[DGFiP] Various changes to make the code C89-compatible

### PR DESCRIPTION
**Pull request by ddeclerck**

### CODE REVIEW

File: src/mlang/backend_compilers/bir_to_dgfip_c.ml [hunk 1]
```diff
@@ -335,7 +336,8 @@ let rec generate_stmt (program : program) (var_indexes : Dgfip_varid.var_id_map)
   | SRuleCall r ->
       let rule = RuleMap.find r program.rules in
       generate_rule_function_header ~definition:false oc rule
-  | SFunctionCall (f, _) -> Format.fprintf oc "%s(irdata);\n" f
+  | SFunctionCall (f, _) -> Format.fprintf oc "%s(irdata);\n" f);
+  Format.fprintf oc "}@]@;"
```
Comment (Keryan-dev) [inline src/mlang/backend_compilers/bir_to_dgfip_c.ml:340]:
```suggestion
  Format.fprintf oc "@]@,}"
```
To have proper indentation on closing bracket.

**Comment by rmonat** [comment]
Thanks a lot David!

I tried forcing `-std=c89` with Clang, I got the following warnings
1) 
```
./desc_static.h.inc:13:3: warning: redefinition of typedef 'T_desc_var' is a C11 feature [-Wtypedef-redefinition]
} T_desc_var;
  ^
./irdata.h:29:27: note: previous definition is here
typedef struct S_desc_var T_desc_var;
```
2)
```
./enchain_static.c.inc:188:8: warning: GCC does not allow variable declarations in for loop initializers before C99 [-Wgcc-compat]
  for (int i = 0; i < nbopd; i++) {
```
3) Complaining there is no stdio.h include for snprintf, math.h include for fmin/fmax.
4) Also, this has not been changed by your PR but it seems the closing brackets of functions are not well-indented. I think `"%a@[<v 2>{@ %a@;return 0;@]@,@\n}@\n"` would do a better job in `generate_rule_function` line 356.
5) Could you confirm all the `generate...` you commented in `generate_c_program` correspond to utilities already written in the `.c.inc`?

**Comment by ddeclerck** [comment]
> I tried forcing `-std=c89` with Clang, I got the following warnings 1)
> 
> ```
> ./desc_static.h.inc:13:3: warning: redefinition of typedef 'T_desc_var' is a C11 feature [-Wtypedef-redefinition]
> } T_desc_var;
>   ^
> ./irdata.h:29:27: note: previous definition is here
> typedef struct S_desc_var T_desc_var;
> ```
> 
> 
> ```
> ./enchain_static.c.inc:188:8: warning: GCC does not allow variable declarations in for loop initializers before C99 [-Wgcc-compat]
>   for (int i = 0; i < nbopd; i++) {
> ```
> 
> 3. Complaining there is no stdio.h include for snprintf, math.h include for fmin/fmax.

I don't get these warnings. What files are you compiling (and how) ?

> 4. Also, this has not been changed by your PR but it seems the closing brackets of functions are not well-indented. I think `"%a@[<v 2>{@ %a@;return 0;@]@,@\n}@\n"` would do a better job in `generate_rule_function` line 356.

I'll look into that (I'm really dumb when it comes to Format :D)

> 5. Could you confirm all the `generate...` you commented in `generate_c_program` correspond to utilities already written in the `.c.inc`?

No, they correspond do things that are no longer required, at least for now (but I prefered to keep them as comment, in case they become required again with the new MPP).

**Comment by rmonat** [comment]
I tried `make test_dgfip_c_backend`, and changed `CC=$(C_COMPILER) -std=c89` in the Makefile `backend_tests`.

Thanks for the clarifications!

**Pull request merged**